electionleaflets/apps/parties/views.py

```diff
@@ -6,13 +6,14 @@
 from django.http import Http404
 from django.views.generic import ListView, TemplateView
 from leaflets.models import Leaflet
-from uk_political_parties.models import Party
 
 
 class PartyList(ListView):
     def get_queryset(self):
-        return Party.objects.annotate(num_leaflets=Count("leaflet")).order_by(
-            "-num_leaflets", "party_name"
+        return (
+            Leaflet.objects.values("ynr_party_id")
+            .annotate(count=Count("ynr_party_id"))
+            .order_by("-count")
```

Comment:
I've not tested it locally, but is this going to work?

The `parties/party_list.html` template is still looking for fields called `party_name` and `num_leaflets`. It looks like the template needs updating to match the query.

Reply:
Yeah, good catch, fixup and test pushed